### PR TITLE
MDEXP-261 Create API for error log handling

### PR DESCRIPTION
**Pull request by illia-borysenko**
MDEXP-261 Create API for error log handling

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: ramls/schemas/logs/recordLog.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "title": {
+      "description": "Record title",
+      "type": "string"
+    },
+    "recordType": {
+      "description": "Record type",
+      "type": "object",
+      "$ref": "../profiles/recordTypes.json"
+    },
+    "affectedRecord": {
```
Comment (andrii-novytskyi) [inline ramls/schemas/logs/recordLog.json:23]: looks like something wrong here, error log contains record uuid, hrid, title, and affected record, that contains record uuid, hrid, title. Is it should be so?

File: ramls/schemas/logs/recordLog.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      "description": "Affected record",
+      "type": "object",
+      "$ref": "recordLog.json"
+    }
+  },
+  "logLevel": [
```
Comment (andrii-novytskyi) [inline ramls/schemas/logs/recordLog.json:29]: `required`

File: src/main/java/org/folio/dao/ErrorLogDao.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   *
+   * @param id       id of {@link ErrorLog} to delete
+   * @param tenantId tenant id
+   * @return future with true is succeeded
+   */
+  Future<Boolean> deleteById(String id, String tenantId);
+
+  /**
+   * Gets {@link ErrorLog} from database
+   *
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/dao/ErrorLogDao.java:50]: Please, format the whole file

File: src/main/java/org/folio/service/logs/ErrorLogService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+   * @param jobExecutionid  id of job execution
+   * @return future with {@link ErrorLogCollection}
+   */
+  Future<ErrorLogCollection> getByJobExecutionId(String jobExecutionid, int offset, int limit, String tenantId);
+
+  /**
+   * Saves {@link ErrorLog} to database
+   *
+   * @param errorLog {@link ErrorLog} to save
+   * @param params         okapi headers and connection parameters
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/service/logs/ErrorLogService.java:22]: the same with formatting this file

**Review by andrii-novytskyi — CHANGES_REQUESTED** [review]

File: ramls/schemas/logs/recordLog.json (hunk 1 above)
Comment (illia-borysenko) [reply ramls/schemas/logs/recordLog.json:23]: yes, we have the following user stories MDEXP-257 (Instances), MDEXP-259 (Holdings), MDEXP-260 (Items), where we should return json response contains information about error for a record and their associated records, like holdings record and the associated instance record, I think it's better to implement such way to return all information in the one response

**Review by andrii-novytskyi — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -410,6 +421,11 @@
       "displayName": "Data Export - get mapping transformation fields collection",
       "description": "Entry point to get mapping transformation fields for the fields"
     },
+    {
+      "permissionName": "ddata-export.error-logs.collection.get",
```
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json:425]: Please correct the typo in permission name

Comment (KVupp) [reply descriptors/ModuleDescriptor-template.json:425]: Also it must be *.item.get. We are getting the logs per ID, not the entire set of logs present

File: ramls/schemas/logs/errorLogCollection.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Collection of Mapping Profiles",
```
Comment (KVupp) [inline ramls/schemas/logs/errorLogCollection.json:3]: Please correct the description

File: src/main/resources/templates/db_scripts/schema.json [hunk 7]
```diff
@@ -59,6 +59,16 @@
           "tOps": "ADD"
         }
       ]
+    },
+    {
+      "tableName": "error_logs",
+      "fromModuleVersion": "mod-data-export-2.2.0",
+      "withMetadata": true,
+      "withAuditing": false,
+      "uniqueIndex": [{
```
Comment (KVupp) [inline src/main/resources/templates/db_scripts/schema.json:68]: we need to have a foreign key as well

Comment (illia-borysenko) [reply src/main/resources/templates/db_scripts/schema.json:68]: Oh, a very good point, thank you!!!

File: ramls/dataExport-logs.raml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    description: Service provides APIs supporting Error logs
+    type:
+      collection:
+        schemaCollection: errorLogCollection
+        exampleCollection: !include samples/logs/errorLogCollection.sample
+    get:
+      is: [
+        pageable,
+        validate,
+        searchable: { description: "with valid searchable fields", example: "jobExecutionId=67dfac11-1caf-4470-9ad1-d533f6360bdd"}
```
Comment (Igor-Gorchakov) [inline ramls/dataExport-logs.raml:38]: please correct the example to do not mention job execution

File: ramls/schemas/logs/errorLog.json [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Error log with all details",
+  "type": "object",
+  "properties": {
+    "id": {
+      "description": "Unique identifier",
+      "$ref": "../../raml-util/schemas/uuid.schema"
+    },
+    "jobExecutionId": {
+      "description": "UUID of the related job execution",
```
Comment (Igor-Gorchakov) [inline ramls/schemas/logs/errorLog.json:11]: fix description

**Review by Igor-Gorchakov — APPROVED** [review]

**Pull request merged**